src/diff/index.js

```diff
@@ -236,18 +236,30 @@ function diffElementNodes(dom, newVNode, oldVNode, context, isSvg, excessDomChil
 
 			let oldHtml = oldProps.dangerouslySetInnerHTML;
 			let newHtml = newProps.dangerouslySetInnerHTML;
-			if ((newHtml || oldHtml) && excessDomChildren==null) {
-				// Avoid re-applying the same '__html' if it did not changed between re-render
-				if (!newHtml || !oldHtml || newHtml.__html!=oldHtml.__html) {
-					dom.innerHTML = newHtml && newHtml.__html || '';
+
+			// During hydration, props are not diffed at all (including dangerouslySetInnerHTML)
+			// @TODO we should warn in debug mode when props don't match here.
+			if (excessDomChildren == null) {
+				if (newHtml || oldHtml) {
+					// Avoid re-applying the same '__html' if it did not changed between re-render
+					if (!newHtml || !oldHtml || newHtml.__html!=oldHtml.__html) {
+						dom.innerHTML = newHtml && newHtml.__html || '';
+					}
 				}
+
+				diffProps(dom, newProps, oldProps, isSvg);
```

Comment:
Does this mean we still diff or is `oldProps` just an empty object? My understanding is that all new props are applied regardless of any existing attributes that may have been present in the dom.

Reply:
ah that's something I didn't think to check for (and one of the failing tests). It seems like my modification here also disables event handler attachment.